src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,90 @@
+package duke;
+
+import java.util.Scanner;
+
+public class Ui implements AutoCloseable {
+    public static final String LONG_LINE = "------------------------------------------------------------";
+    // Default indentation is 8 whitespaces
+    public static final char DEFAULT_INDENT_CHARACTER = '\t';
+    public static final int DEFAULT_INDENT_COUNT = 1;
+    // Used for internal 2nd-level indentation (e.g. print task)
+    public static final String INTERNAL_INDENT = "\t";
+
+    protected String indent;
+    protected Scanner scanner;
+
+    public Ui() {
+        this(DEFAULT_INDENT_CHARACTER, DEFAULT_INDENT_COUNT);
+    }
+
+    public Ui(char indentCharacter, int indentCount) {
+        indent = new String(new char[indentCount]).replace('\0', indentCharacter);
+        scanner = new Scanner(System.in);
+    }
+
+    @Override
+    public void close() {
+        scanner.close();
+        scanner = null;
+    }
+
+    public String read() {
+        if (scanner == null || !scanner.hasNextLine()) {
+            return null;
+        }
+        return scanner.nextLine();
+    }
+
+    public void print(String text) {
+        String[] lines = text.split("\n");
+        for (String line : lines) {
+            System.out.println(indent + line);
+        }
+    }
+
+    // Print a message for a successful insertion of task
```

Comment:
Use Javadoc comment to describe the purpose of a method

Reply:
Thanks for the advice. This has been fixed in the latest commit.